### PR TITLE
renames CLI flag from --storage.trace.maintenance-cycle to --storage.trace.blocklist_poll

### LINKED ISSUE #862: Rename CLI flag: --storage.trace.maintenance-cycle to --storage.trace.blocklist_poll
**Describe the bug**
The CLI has a flag `--storage.trace.maintenance-cycle`
https://github.com/grafana/tempo/blob/c61b67851b576a673176582f3388eb3c7f0c7145/modules/storage/config.go#L34

But in the YAML config this flag corresponds with `storage.trace.blocklist_poll`
https://github.com/grafana/tempo/blob/9485f96bf7042a3b0e98d93a7c3aa0e42287074c/tempodb/config.go#L30-L31

Changing this will be a breaking change.

**Expected behavior**
Name of YAML config options and CLI flags match with each other.

### PR DESCRIPTION
**Pull request by mritunjaysharma394**
Signed-off-by: Mritunjay Sharma <mritunjaysharma394@gmail.com>

**What this PR does**:
This PR renames` --storage.trace.maintenance-cycle` to` --storage.trace.blocklist_poll`
**Which issue(s) this PR fixes**:
Fixes #862 

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by joe-elliott** [comment]
We will hold off on merging this until we cut 1.1 (next week), but thanks for the fix!

One nit: can we list this as a breaking change in the change log?

**Comment by mritunjaysharma394** [comment]
Thanks a lot @joe-elliott for the review yes I fixed it to breaking change in the change log, thank you so much :blush:

**Review by annanay25 — COMMENTED** [review]
1.1 has been cut, can you please rebase your PR over main @mritunjaysharma394? Thanks for the contribution!

**Review by joe-elliott — APPROVED** [review]

**Comment by mritunjaysharma394** [comment]
> 1.1 has been cut, can you please rebase your PR over main @mritunjaysharma394? Thanks for the contribution!

Hi @annanay25 , thanks for the review, I have tried to do it, let me know if it works, thank you so much :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -30,6 +30,7 @@
 * [ENHANCEMENT] Add `gen index` and `gen bloom` commands to tempo-cli. [#903](https://github.com/grafana/tempo/pull/903) (@annanay25)
 
 ## v1.1.0 / 2021-08-26
+* [CHANGE] Renamed CLI flag from `--storage.trace.maintenance-cycle` to `--storage.trace.blocklist_poll`. This is a **breaking change**  [#897](https://github.com/grafana/tempo/pull/897) (@mritunjaysharma394)
```
Comment (joe-elliott) [inline CHANGELOG.md:33]: one last change. we need to move this to the section above :point_up: . This change is not in 1.1. It is currently unreleased.

Comment (mritunjaysharma394) [reply CHANGELOG.md:33]: oops, yes, I am just fixing it, thanks a lot!

**Comment by joe-elliott** [comment]
Thank you! I will merge as soon as CI passes.

**Comment by mritunjaysharma394** [comment]
> Thank you! I will merge as soon as CI passes.

Thank you so much for all the help! Looking forward to learn and contribute more :smile:

**Pull request merged**